dev-libs/boost/boost-1.88.0.ebuild

```diff
@@ -0,0 +1,459 @@
+		# there's no support for special options like O_LARGEFILE.
+		# Thus, LFS must be disabled.
+		#
+		# On other systems, we need to enable LFS explicitly for 64-bit
+		# offsets on 32-bit hosts (#894564)
+		append-lfs-flags
+	fi
+}
+
+multilib_src_compile() {
+	ejam \
+		--prefix="${EPREFIX}"/usr \
+		"${OPTIONS[@]}" || die
+
+	if multilib_native_use tools; then
+		pushd tools >/dev/null || die
+		ejam \
+			--prefix="${EPREFIX}"/usr \
+			"${OPTIONS[@]}" \
+			|| die "Building of Boost tools failed"
+		popd >/dev/null || die
+	fi
+}
+
+multilib_src_test() {
+	# create custom options for tests based on the build settings
+	TEST_OPTIONS=("${OPTIONS[@]}")
+
+	# dial down log output - the full b2 command used to compile & run
+	# a test suite will be printed by ejam and can be used to build
+	# and run the tests in a test suite's directory.
+	TEST_OPTIONS=("${TEST_OPTIONS[@]/-d+2/-d0}")
+
+	# The following libraries do not compile or fail their tests:
+	local libs_excluded=(
+		# fails to use std::reverse_copy
+		"algorithm"
+		# test output comparison failure
+		"config"
+		# "C++03 support was deprecated in Boost.Chrono 1.82" ??
+		"contract"
+		# passes but takes >1 hour to build (enabled with USE=test-full)
+		"geometry"
+		# tries to use MPI even with use=-mpi (enabled with USE=mpi)
+		"graph_parallel"
+		# undefined reference to `boost::math::concepts::real_concept boost::math::bernoulli_b2n<boost::math::concepts::real_concept>(int)
+		"math"
+		# assignment of read-only member 'gauss::laguerre::detail::laguerre_l_object<T>::order'
+		"multiprecision"
+		# uint8_t is not a member of std
+		"mysql"
+		# PyObject* boost::parameter::python::aux::unspecified_type():
+		#   /usr/include/python3.13/object.h:339:30: error: lvalue required as left operand of assignment
+		"parameter_python"
+		# scope/lambda_tests22.cpp(27): test 'x == 1' failed in function 'int main()'
+		"phoenix"
+		# Unable to find file or target named (yes, really)
+		"predef"
+		# AttributeError: property '<unnamed Boost.Python function>' of 'X' object has no setter
+		"python"
+		# vec_access.hpp:95:223: error: static assertion failed: Boost QVM static assertion failure
+		"qvm"
+		# regex_timer.cpp:19: ../../../boost/timer.hpp:21:3: error: #error This header is
+		#   deprecated and will be removed. (You can define BOOST_TIMER_ENABLE_DEPRECATED to suppress
+		#   this error.)
+		"regex"
+		# in function `boost::archive::tmpnam(char*)': test_array.cpp:(.text+0x108):
+		#   undefined reference to `boost::filesystem::detail::unique_path(...)'
+		"serialization"
+		# TuTestMain.cpp(22) fatal error: in "test_main_caller( argc_ argv )":
+		#   std::runtime_error: Event was not consumed!
+		"statechart"
+		# erase_tests.cpp:(.text+0x44cce): undefined reference to
+		#   tbb::detail::r1::execution_slot(tbb::detail::d1::execution_data const*)
+		"unordered"
+		# t_5_007.cpp(22): error: could not find include file: boost/version.hpp
+		"wave"
+	)
+
+	if use mpi; then
```

Comment:
IMO it's nicer to do `if ! use mpi` and add it to the exclude list, and drop it from the array above (ditto below).

Reply:
Switched both mpi and test-full to negative checks and adding to the exclusion list.
I also added mpi to the exclusion list when USE=-mpi. I think the tests passed even with -mpi but that's just nonsense.